Source/ARTPushChannel.m

```diff
@@ -0,0 +1,122 @@
+//
+//  ARTPushChannel.m
+//  Ably
+//
+//  Created by Ricardo Pereira on 07/02/2017.
+//  Copyright © 2017 Ably. All rights reserved.
+//
+
+#import "ARTPushChannel.h"
+#import "ARTHttp.h"
+#import "ARTLog.h"
+#import "ARTChannel.h"
+#import "ARTJsonLikeEncoder.h"
+
+@implementation ARTPushChannel {
+    id<ARTHTTPAuthenticatedExecutor> _httpExecutor;
+    __weak ARTLog *_logger;
+    __weak ARTChannel *_channel;
+}
+
+- (instancetype)init:(id<ARTHTTPAuthenticatedExecutor>)httpExecutor withChannel:(ARTChannel *)channel {
+    if (self == [super self]) {
+        _httpExecutor = httpExecutor;
+        _logger = [httpExecutor logger];
+        _channel = channel;
+    }
+    return self;
+}
+
+- (void)subscribeForDevice:(ARTDeviceId *)deviceId {
+    NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:[NSURL URLWithString:@"/push/channelSubscriptions"]];
+    request.HTTPMethod = @"POST";
+    request.HTTPBody = [[_httpExecutor defaultEncoder] encode:@{
+        @"deviceId": deviceId,
+        @"channel": _channel.name,
+    }];
+    [request setValue:[[_httpExecutor defaultEncoder] mimeType] forHTTPHeaderField:@"Content-Type"];
+
+    [_logger debug:__FILE__ line:__LINE__ message:@"subscribe notifications for device %@ in channel %@", deviceId, _channel.name];
+    [_httpExecutor executeRequest:request withAuthOption:ARTAuthenticationOn completion:^(NSHTTPURLResponse *response, NSData *data, NSError *error) {
+        if (response.statusCode == 200 /*OK*/) {
+            return;
+        }
+        if (error) {
+            [_logger error:@"%@: subscribe notifications for device %@ in channel %@ failed (%@)", NSStringFromClass(self.class), deviceId, _channel.name, error.localizedDescription];
+        }
+        else {
+            [_logger error:@"%@: subscribe notifications for device %@ in channel %@ failed with status code %ld", NSStringFromClass(self.class), deviceId, _channel.name, (long)response.statusCode];
+        }
+    }];
+}
+
+- (void)subscribeForClientId:(NSString *)clientId {
```

Comment:
> In the IDL it is implicit. But I don't know, I guess it can work like Presence#enterClient?

Not sure I see a use case for a client device to specify `clientId` other than the one implicit in the token.  Servers on the other hand would do this frequently, is this API intended for server-side access as well?